### PR TITLE
Update documentation to refer to `archiver` and not `scrapers` or `zenodo-storage`

### PR DESCRIPTION
**Pull request by jdangerx**
# PR Overview

Exactly what it says on the tin.

One hiccup - I'm not completely sure if the changes in `Annual Updates` are strictly correct - @zschira maybe you have better insight?

# PR Checklist

- [x] Merge the most recent version of the branch you are merging into (probably `dev`).
- [ ] All CI checks are passing. [Run tests locally to debug failures](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#running-tests-with-tox)
- [x] Make sure you've included good docstrings.
- [x] Update the [release notes](https://catalystcoop-pudl.readthedocs.io/en/latest/release_notes.html) and reference reference the PR and related issues.

### CODE REVIEW

File: docs/release_notes.rst [hunk 1]
```diff
@@ -34,6 +34,14 @@ Data Coverage
   * :ref:`retained_earnings_ferc1`, see issue :issue:`1811` & PR :pr:`2155`.
   * :ref:`cash_flow_ferc1`, see issue :issue:`1821` & PR :pr:`2184`
 
+Deprecations
+^^^^^^^^^^^^
+
+* Replace references to deprecated ``pudl-scrapers`` and
+  ``pudl-zenodo-datastore`` repositories with references to `pudl-archiver
+  <https://www.github.com/catalyst-cooperative/pudl-archiver>`__ repository in
+  :doc:`intro`, :doc:`dev/datastore`, and :doc:`dev/annual_updates`. See :pr:`2190`.
```
Comment (jdangerx) [inline docs/release_notes.rst:43]: Not sure how to link to an issue from another repository but I figured this specific one wasn't super important.

**Comment by zaneselvans** [comment]
Hey, I think this probably overlaps with the draft PR #2089 from @aesharpe which we should really review and merge.

**Comment by aesharpe** [comment]
Ya that would help. @zschira 's new scraper archiver stuff hasn't been added to that PR or the docs because (I think) he is waiting to get the automation up and running, but I think it would make sense to explain what it does now and that we are working on automation.

**Comment by jdangerx** [comment]
@aesharpe @zschira want to grab some time this week to crunch through all those documentation updates on a call?

File: docs/dev/datastore.rst [hunk 2]
```diff
@@ -58,122 +57,8 @@ present locally.
...
-``zenodo_store.py --initialize --verbose --sandbox``
-
-If successful, the DOI and url for your archive will be printed.  You will
-need to visit the url to review and publish the Zenodo archive before it can
-be used.
-
-If you lose track of the DOI, you can look up the archive on Zenodo using the
-UUID from ``zs.metadata``.
-
-Prepare the Datastore
```
Comment (zschira) [inline docs/dev/datastore.rst:155]: This section on the `Datastore` should probably remain. the `Datastore` is contained in PUDL, and hasn't changed (yet)

File: docs/release_notes.rst (hunk 1 above)
Comment (zschira) [reply docs/release_notes.rst:43]: I'd agree it's not too important to link the specific issue

File: docs/dev/annual_updates.rst [hunk 3]
```diff
@@ -22,34 +22,23 @@ into integrating the 2020 data.
 
 1. Obtain Fresh Data
```
Comment (zschira) [inline docs/dev/annual_updates.rst:23]: These changes look good to, but we will need to remember to update this again once automated archives are working

**Review by zschira — CHANGES_REQUESTED** [review]
Just one minor change.

**Comment by jdangerx** [comment]
@zschira mind taking another look?

**Pull request merged**